apps/frontend/src/routes/+layout.svelte

```diff
@@ -17,42 +22,74 @@
 		Dropdown,
 		DropdownHeader,
 		DropdownDivider,
-		Avatar,
-		DarkMode
+		Avatar
 	} from 'flowbite-svelte';
 
 	import LanguageSwitcher from '$lib/components/LanguageSwitcher.svelte';
 	import SignInButton from '$lib/auth/components/SignInButton.svelte';
 
 	let { children } = $props();
 
-	let isDarkMode = $state(true);
+	let isDarkMode = $state(false);
 
-	// Auto dark mode based on browser preference
+	// Initialize theme immediately
 	onMount(() => {
-		const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
-
-		const updateTheme = (e: MediaQueryList | MediaQueryListEvent) => {
-			if (e.matches) {
-				document.documentElement.classList.add('dark');
-				isDarkMode = true;
-			} else {
-				document.documentElement.classList.remove('dark');
-				isDarkMode = false;
-			}
-		};
+		// Check localStorage first, then fall back to system preference
+		const savedTheme = localStorage.getItem('theme');
+		const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+
+		isDarkMode = savedTheme === 'dark' || (!savedTheme && prefersDark);
+
+		if (isDarkMode) {
+			document.documentElement.classList.add('dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+		}
+
+		// Listen for system preference changes only if no saved preference
+		if (!savedTheme) {
+			const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+			const updateTheme = (e: MediaQueryListEvent) => {
+				isDarkMode = e.matches;
+				if (e.matches) {
+					document.documentElement.classList.add('dark');
+				} else {
+					document.documentElement.classList.remove('dark');
+				}
+			};
+
+			mediaQuery.addEventListener('change', updateTheme);
+
+			return () => {
+				mediaQuery.removeEventListener('change', updateTheme);
+			};
+		}
+	});
 
-		// Set initial theme
-		updateTheme(mediaQuery);
+	function toggleTheme() {
+		isDarkMode = !isDarkMode;
+		if (isDarkMode) {
+			document.documentElement.classList.add('dark');
+			localStorage.setItem('theme', 'dark');
+		} else {
+			document.documentElement.classList.remove('dark');
+			localStorage.setItem('theme', 'light');
+		}
+	}
+</script>
 
-		// Listen for changes
-		mediaQuery.addEventListener('change', updateTheme);
+<svelte:head>
+	<script>
+		(function () {
+			const savedTheme = localStorage.getItem('theme');
+			const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
 
-		return () => {
-			mediaQuery.removeEventListener('change', updateTheme);
-		};
-	});
-</script>
+			if (savedTheme === 'dark' || (!savedTheme && prefersDark)) {
+				document.documentElement.classList.add('dark');
+			}
+		})();
+	</script>
+</svelte:head>
```

Comment:
This looks redundant with the code in `onMount()`. Do we really need both? If so, can we make it more DRY?

If not, let's remove whichever one we don't need.